hbase-server/src/main/java/org/apache/hadoop/hbase/io/FSDataInputStreamWrapper.java

```diff
@@ -270,39 +269,23 @@ public void unbuffer() {
       if (this.instanceOfCanUnbuffer == null) {
         // To ensure we compute whether the stream is instance of CanUnbuffer only once.
         this.instanceOfCanUnbuffer = false;
-        Class<?>[] streamInterfaces = streamClass.getInterfaces();
-        for (Class c : streamInterfaces) {
-          if (c.getCanonicalName().toString().equals("org.apache.hadoop.fs.CanUnbuffer")) {
-            try {
-              this.unbuffer = streamClass.getDeclaredMethod("unbuffer");
-            } catch (NoSuchMethodException | SecurityException e) {
-              if (isLogTraceEnabled) {
-                LOG.trace("Failed to find 'unbuffer' method in class " + streamClass
-                    + " . So there may be a TCP socket connection "
-                    + "left open in CLOSE_WAIT state.", e);
-              }
-              return;
-            }
-            this.instanceOfCanUnbuffer = true;
-            break;
-          }
+        if (CanUnbuffer.class.isAssignableFrom(streamClass)) {
+          this.unbuffer = (CanUnbuffer) wrappedStream;
+          this.instanceOfCanUnbuffer = true;
         }
       }
       if (this.instanceOfCanUnbuffer) {
         try {
-          this.unbuffer.invoke(wrappedStream);
-        } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+          this.unbuffer.unbuffer();
+        } catch (IllegalArgumentException | UnsupportedOperationException e){
```

Comment:
I added the try catch here, because some UT failed such as TestHFile. The failed message is "this stream org.apache.hadoop.fs.BufferedFSInputStream does not support unbuffering." check the code the BufferedFSInputStream is not implements the unbuffer, through FSDataInputStream implements unbuffer but the InputStream paramenter maybe NOT, so we should catch it.

Reply:
The unbuffer here is an CanUnbuffer instance so why we could still get this exception? And for BufferedFSInputStream, it is not a sub class of CanUnbuffer? Could you please check the UT again?

Reply:
@Apache9  My fault! We do NOT need catch the IllegalAccessException. Yes, the BufferedFSInputStream is NOT a sub class of CanUnbuffer, checked in the hadoop-common project. So need to catch the UnsupportedOperationException, because the BufferedFSInputStream is NOT a sub class of CanUnbuffer. Commit a change remove catch IllegalAccessException. Thanks for your review and point out my fault.

Reply:
@Apache9 Could you help review again? Thanks!